core/src/main/java/hudson/model/listeners/ItemListener.java

```diff
@@ -57,6 +58,18 @@ public void onCreated(Item item) {
     }
 
     /**
+     * Called before a job is copied into a new parent, providing the ability to veto the copy operation before it
+     * starts.
+     *
+     * @param src the item being copied
+     * @param parent the proposed parent
+     * @throws Failure to veto the operation.
+     * @since TODO
+     */
+    public void onCheckCopy(Item src, ItemGroup parent) throws Failure {
```

Comment:
🐛 Failure is the wrong exception class. Javadoc says that it "Represents an error induced by user, encountered during HTTP request processing. ", but from what I see it is not the only use-case for this listener. E.g. it may be used in CLI commands for copying jobs